### PR TITLE
Implement working smoke-test

### PR DESCRIPTION
**Pull request by colinbruce**
## What?
This creates a SmokeTest class, for use case one, that can be run by opening the rails console and running
```ruby 
  SmokeTest.call
```
This is used by the also implemented API endpoint.  In non-live environments, it can be called via browser or Curl using `https://~host~/smoke-tests`
It adds a timed job that will run the smoke test at 8:30 each day to ensure the Sandbox we hit has not changed its format

## Why?

Two reasons:
- because we can't guarantee that we will regularly access the sandbox - this will execute a test once a day to ensure that the sandbox is working as expected and;
- because I have no confidence that the endpoints won't suddenly change, this should act as an early warning system if an endpoint suddenly changes e.g. from v1 to v2

 ### ToDo
- Add additional smoke tests for other use-cases, this will require setting up more sandbox data and extending the secrets in this repo

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -149,6 +149,17 @@ jobs:
...
+  smoke_tests:
+    executor: cloud-platform-executor
+    steps:
+      - checkout
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - *decrypt_secrets
+      - *authenticate_k8s
+      - run:
+          name: Run curl
```
Comment (willc-work) [inline .circleci/config.yml:161]: this could be a more useful desciption

File: .circleci/config.yml [hunk 2]
```diff
@@ -312,6 +325,16 @@ workflows:
 #          requires:
 #            - hold_production
 
+  smoke-test:
+    triggers:
+      - schedule:
+          cron: "30 8 * * *"
```
Comment (willc-work) [inline .circleci/config.yml:331]: what happens if it fails? I am sure it is graceful bit would there be any point running it sat and sun if no one is going to look or be able to respond?

File: spec/requests/smoke_test_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  let(:expected_result) { { smoke_test_result: smoke_test_pass? }.to_json }
+
+  describe '#call' do
+    context 'when the smoke test passes' do
+      let(:smoke_test_pass?) { true }
+      it { expect(response.body).to eql expected_result }
+      it { expect(response.status).to eql 200 }
+    end
+
+    context 'when the smoke test failes' do
```
Comment (willc-work) [inline spec/requests/smoke_test_controller_spec.rb:18]: sp error in failes

**Review by willc-work — APPROVED** [review]
minor comments, looks good @

File: .circleci/config.yml (hunk 2 above)
Comment (colinbruce) [reply .circleci/config.yml:331]: At some point I'll set up a new slack channel and implement slack orb notifications 👍

Comment (colinbruce) [reply .circleci/config.yml]: And I've added that now...

**Review by willc-work — APPROVED** [review]

**Pull request merged**